inc/modules/export/class-table.php

```diff
@@ -0,0 +1,451 @@
+		foreach ( $groups as $group ) {
+			$i = 1;
+			foreach ( $group as $filepath ) {
+				if ( $i > $keep_the_last ) {
+					unlink( $filepath );
+				}
+				++$i;
+			}
+		}
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function getLatestExports() {
+		$files = [];
+		foreach ( $this->getFiles() as $filepath ) {
+			$stat = stat( $filepath );
+			$file = basename( $filepath );
+			$id = $this->getTinyHash( $file );
+			$files[] = [
+				'ID' => $id,
+				'file' => $file,
+				'format' => $this->getFormat( $file ),
+				'size' => \Pressbooks\Utility\format_bytes( $stat['size'] ),
+				'pin' => $this->hasPin( $id ) ? 1 : 0,
+				'exported' => date_i18n( 'Y-m-d H:i', $stat['mtime'] ),
+			];
+		}
+		return $files;
+	}
+
+	/**
+	 * @param string $file
+	 *
+	 * @return string
+	 */
+	protected function getCssClass( $file ) {
+		$file_extension = substr( strrchr( $file, '.' ), 1 );
+		switch ( $file_extension ) {
+			case 'epub':
+				$pre_suffix = strstr( $file, '._3.epub' );
+				break;
+			case 'pdf':
+				$pre_suffix = strstr( $file, '._print.pdf' );
+				break;
+			case 'html':
+				$pre_suffix = strstr( $file, '.-htmlbook.html' );
+				break;
+			case 'xml':
+				$pre_suffix = strstr( $file, '._vanilla.xml' );
+				break;
+			default:
+				$pre_suffix = false;
+		}
+		if ( 'html' === $file_extension && '.-htmlbook.html' === $pre_suffix ) {
+			$file_class = 'htmlbook';
+		} elseif ( 'html' === $file_extension && false === $pre_suffix ) {
+			$file_class = 'xhtml';
+		} elseif ( 'xml' === $file_extension && '._vanilla.xml' === $pre_suffix ) {
+			$file_class = 'vanillawxr';
+		} elseif ( 'xml' === $file_extension && false === $pre_suffix ) {
+			$file_class = 'wxr';
+		} elseif ( 'epub' === $file_extension && '._3.epub' === $pre_suffix ) {
+			$file_class = 'epub3';
+		} elseif ( 'pdf' === $file_extension && '._print.pdf' === $pre_suffix ) {
+			$file_class = 'print-pdf';
+		} else {
+			/**
+			 * Map custom export format file extensions to their CSS class.
+			 *
+			 * For example, here's how one might set the CSS class for a .docx file:
+			 *
+			 * add_filter( 'pb_get_export_file_class', function ( $file_extension ) {
+			 *    if ( 'docx' == $file_extension ) {
+			 *        return 'word';
+			 *    }
+			 *    return $file_extension;
+			 * } );
+			 *
+			 * @since 3.9.8
+			 *
+			 * @param string $file_extension
+			 */
+			$file_class = apply_filters( 'pb_get_export_file_class', $file_extension );
+		}
+		return $file_class;
+	}
+
+	/**
+	 * @param string $file
+	 * @param string $size
+	 *
+	 * @return string
+	 */
+	protected function getIcon( $file, $size = 'large' ) {
+		$file_class = $this->getCssClass( $file );
+		$html = "<div class='export-file-icon {$size} {$file_class}' title='" . esc_attr( $file ) . "'></div>";
+		return $html;
+	}
+
+	/**
+	 * @param string $file
+	 *
+	 * @return string
+	 */
+	protected function getFormat( $file ) {
+		$file_class = $this->getCssClass( $file );
+		return ucfirst( $file_class );
```

Comment:
We should use something like this instead (moving the functionality into core would mean we could also remove it from McLuhan): https://github.com/pressbooks/pressbooks-book/blob/dev/inc/helpers/namespace.php#L76-L107